src/components/Layout.tsx

```diff
@@ -0,0 +1,43 @@
+import { AppBar, Toolbar, MenuItem, Typography, Button } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { Link, Outlet } from 'react-router-dom';
+import i18n from '../i18n';
+
+export default function Layout() {
+  const { t } = useTranslation();
+
+  const changeLanguage = (lng: string | undefined) => {
```

Comment:
Not related with this PR, but I'm wondering, why is the `lng` typed as `string | undefined`? You are passing only `en` and `pl`, so I would prefer, to be as precise as possible (using the string literals), or at least only `string`. Is the `undefined` needed here? :)